### PR TITLE
[FORM] fix post_max_size_message translation (alt. 2)

### PR DESCRIPTION
**Pull request by DavidBadura**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15479, #18543 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            return $this->translator->trans($errorMessage, array(), $this->translationDomain);
+        });
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getExtendedType()
+    {
+        return FormType::class;
```
Comment (HeahDude) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: Should be `'form'` as it targets 2.7 and to be changed in 2.8.

File: src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+}
```
Comment (HeahDude) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: Missing a return here

File: src/Symfony/Component/Form/Tests/Extension/Validator/Type/UploadValidatorExtensionTest.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+        $resolver = new OptionsResolver();
+        $resolver->setDefault('post_max_size_message', 'old max {{ max }}!');
+
+        $extension->configureOptions($resolver);
+        $options = $resolver->resolve();
+
+        $this->assertEquals('translated max {{ max }}!', $options['post_max_size_message']);
+    }
+}
```
Comment (HeahDude) [inline src/Symfony/Component/Form/Tests/Extension/Validator/Type/UploadValidatorExtensionTest.php]: here as well

File: src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php (hunk 2 above)
Comment (HeahDude) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: @fabpot Can't we add this check in the CS fixer?

**Comment by DavidBadura** [comment]
comments addressed

File: src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $translator = $this->translator;
+
+        $resolver->setNormalizer('post_max_size_message', function (Options $options, $errorMessage) use ($translator) {
+            return $translator->trans($errorMessage, array(), $this->translationDomain);
```
Comment (ro0NL) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: Why not just use `$this->translator` directly as done with `$this->translationDomain`?

Comment (jvasseur) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: Symfony 2.7 need to be compatible with php5.3 so we can't use `$this` in closures, the `$this->translationDomain` needs to be replaced.

Comment (DavidBadura) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: oh, that's wrong. $this in anonymous function you can use starting 5.4.

Comment (DavidBadura) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: fixed

Comment (ro0NL) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: :+1:

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/form.xml [hunk 5]
```diff
@@ -189,5 +189,10 @@
         <service id="form.type_extension.submit.validator" class="Symfony\Component\Form\Extension\Validator\Type\SubmitTypeValidatorExtension">
             <tag name="form.type_extension" alias="submit" />
         </service>
+        <service id="form.type_extension.submit.validator" class="Symfony\Component\Form\Extension\Validator\Type\UploadValidatorExtension">
```
Comment (HeahDude) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/form.xml]: The id need to be changed

Comment (DavidBadura) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/form.xml]: fixed

File: src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+ */
+class UploadValidatorExtension extends AbstractTypeExtension
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var null|string
```
Comment (HeahDude) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: I guess you can remove phpdocs on both private properties, those for the constructor are enough

File: src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $translator = $this->translator;
+        $translationDomain = $this->translationDomain;
+
+        $resolver->setNormalizer(
+            'post_max_size_message',
```
Comment (HeahDude) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: Can be inlined

File: src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php (hunk 6 above)
Comment (DavidBadura) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: done

File: src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php (hunk 7 above)
Comment (DavidBadura) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: then there are 140 characters in a row... this is to much ;-)

Comment (ro0NL) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]:
```php
        $resolver->setNormalizer('post_max_size_message', function (Options $options, $errorMessage) use ($translator, $translationDomain) {
            return $translator->trans($errorMessage, array(), $translationDomain);
        });
```

140 exactly :D i could live with it..

Comment (ro0NL) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: Is it? PSR-wise there's no hard limit, soft limit is 120 though...

Comment (WouterJ) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: Symfony doesn't do line wrapping and has no limit, soft, hard, anything.

Comment (fabpot) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: Anyway, we don't care about those limits in Symfony.

Comment (ro0NL) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: So inline then?

Comment (DavidBadura) [inline src/Symfony/Component/Form/Extension/Validator/Type/UploadValidatorExtension.php]: changed

**Comment by fabpot** [comment]
It seems weird to create an extension for this, but  I think I like this implementation more than the one in #18543, just because it is less intrusive as a bug fix for 2.7.

👍 
ping @symfony/deciders

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/form.xml [hunk 8]
```diff
@@ -189,5 +189,10 @@
         <service id="form.type_extension.submit.validator" class="Symfony\Component\Form\Extension\Validator\Type\SubmitTypeValidatorExtension">
             <tag name="form.type_extension" alias="submit" />
         </service>
+        <service id="form.type_extension.upload.validator" class="Symfony\Component\Form\Extension\Validator\Type\UploadValidatorExtension">
+            <tag name="form.type_extension" alias="form" />
```
Comment (HeahDude) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/form.xml]: If I'm not wrong `alias` is not used anymore

Comment (HeahDude) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/form.xml]: Sorry I forgot this target 2.7 :)

**Comment by HeahDude** [comment]
👍

**Comment by fabpot** [comment]
Thank you @DavidBadura.

**Pull request merged**